src/frontend/qt_sdl/main.cpp

```diff
@@ -507,6 +516,14 @@ void EmuThread::run()
                     lasttick = SDL_GetTicks();
                 }
             }
+            
+            // This piece of code is used in order for the Rumble Pak
+            // implementation to take into account variable framerates...
+            if (Input::isFrameDone)
```

Comment:
I don't know if we even need `IsFrameDone`, since it's set to true whenever rumble is set. You can probably just call `StopRumble()` after every frame based on your current implementation. Anyway the variable name doesn't make sense to me, since it seems to indicate that rumble should be stopped once the frame is done (rather than the frame actually being done).

Besides that, if you've tested it and it feels right to you I'm fine with that implementation. I don't want to delay this until I get around to comparing it to real hardware, but I'd like a TODO added here to remind developers that testing is needed.